helix-rest/src/main/java/org/apache/helix/rest/server/resources/AbstractResource.java

```diff
@@ -130,6 +130,10 @@ private String errorMsgToJson(String error) {
   }
 
   protected Response JSONRepresentation(Object entity) {
+    if (entity == null) {
+      // return empty content when entity is null
+      return OK("");
+    }
```

Comment:
1) Why is this necessary all of a sudden?
2) Shouldn't we do checks like this in toJson function?

Reply:
Because I noticed getting the data from a non-existed path gives me the string result 
"null". It should not be the expected

Reply:
I don't understand your comment. What's wrong with returning null when the ZNode doesn't exist?

If you return return OK(""), then doesn't that mean that there exists a ZNode with empty content?

Reply:
Currently, we're returning string "null" if the data path doesn't exist. One can interpret as 1. the content in the data path is "null" 2. The data path doesn't exist. Isn't it confusing?

Reply:
This should not be the right behavior. We need to return 400 for path not found instead of 200 for OK.

Reply:
Fixed. Returning 404 when the path doesn't exist